transformations/r-plain/index.md

```diff
@@ -0,0 +1,239 @@
+a [sample CSV file](/transformations/r/source.csv) locally, take the following steps:
+
+- Put the R code into a file, for instance, script.R in the working directory.
+- Put all tables from the input mapping inside the `in/tables` subdirectory of the working directory.
+- Place the binary files (if using any) inside the `in/user` subdirectory of the working directory, and make sure
+  that their name has no extension.
+- Store the result CSV files inside the `out/tables` subdirectory.
+
+Use this sample script:
+{% highlight r %}
+data <- read.csv(file = "in/tables/source.csv");
+
+df <- data.frame(
+col1 = paste0(data$first, 'ping'),
+col2 = data$second * 42
+)
+write.csv(df, file = "out/tables/result.csv", row.names = FALSE)
+{% endhighlight %}
+
+A complete example of the above is attached below in [data.zip](/transformations/r/data.zip).
+Download it and test the script in your local R installation. The `result.csv` output file will be created.
+This script can be used in your transformations without any modifications.
+All you need to do is
+
+- create a table in Storage by uploading the [sample CSV file](/transformations/r/source.csv),
+- create an input mapping from that table, setting its destination to `source` (as expected by the R script),
+- create an output mapping from `result.csv` (produced by the R script) to a new table in your Storage,
+- copy & paste the above script into the transformation code, and finally,
+- save and run the transformation.
+
+### Events and Output
+It is possible to output informational and debug messages from the R script simply by printing them out.
+The following R script:
+
+{% highlight r %}
+
+print('doing something')
+Sys.sleep(3)
+print('doing something else')
+Sys.sleep(3)
+write('still doing something', stdout())
+Sys.sleep(3)
+write('error message', stderr())
+Sys.sleep(3)
+app$logInfo("information")
+Sys.sleep(3)
+app$logError("error")
+Sys.sleep(3)
+TRUE
+{% endhighlight %}
+
+produces the following events in the transformation job:
+
+![Screenshot - Script Events](/transformations/r-plain/events.png)
+
+The `app$logInfo` and `app$logError` functions are also internally available; they can be useful if you need to know 
+the precise server time of when an event occurred. The standard event timestamp in job events is the time when the event was received
+converted to the local time zone.
+
+### Going Further
+The above steps are usually sufficient for daily development and debugging of moderately complex R transformations,
+although they do not reproduce the transformation execution environment exactly. To create a development environment
+with the exact same configuration as the transformation environment, use [our Docker image](https://developers.keboola.com/extend/docker/running/#running-transformations).
+
+## Examples
+There are more in-depth examples dealing with
+
+- [array splitting](/transformations/r/array-splitter/),
```

Comment:
yes, I believe that they are too.  will double check.

What about the image references to the legacy transformation (I've kept what still applies).  I suppose it would be better to also copy them into this new dir eh

Reply:
just move it, you can put a redirect in the header https://keboola.atlassian.net/wiki/spaces/KB/pages/82935879/Public+documentation#Changes%2C-Redirects so you don't even have to change the old transformations

Reply:
ah, it doesn't perfectly transfer, needs replacing a couple of images that show the old transformations (1 on each page).  the concept transfers though

Reply:
how about I make a copy of them and edit accordingly?  rather than moving?

Reply:
whatever fits